### ISSUE TITLE
Depth-Offset: A Single Value is both Constant Offset and Slope Bias

### ISSUE THREAD

**Issue by ArsThaumaturgis** [panda3d/panda3d#1157]
## Description
The current "setDepthOffset" method takes a single parameter. This parameter is then used for both the constant offset _and_ the slope-scaled bias. The result is that, at steep angles, the resultant offset can become high enough that the affected geometry can be seen through occluding objects at an unexpected distance.

See here:
![Screenshot from 2021-04-27 17-31-10](https://user-images.githubusercontent.com/36933600/116395233-26e89d80-a824-11eb-85e0-0ddd66cce6d8.png)

The black line is produced by a black plane at near right-angles to the camera, and placed behind the white plane.

Note that this screenshot used a depth-offset of 5, in order to exaggerate the effect, I believe.

## Steps to Reproduce
Simply put, load the following model into PView, centre the view on it, and then move and angle the camera until you see a black line flickering through the white surface.

[depthTest.zip](https://github.com/panda3d/panda3d/files/6391423/depthTest.zip)
(The model has been zipped simply so that the bug-tracker will take it.)

## Environment
 * Operating system: Ubuntu 18.04.5
 * System architecture: 64-bit
 * Panda3D version: 1.10.9
 * Installation method: Pip, I do believe.
 * Python version (if using Python): 3.6.9
 * Compiler (if using C++): N/A

**Comment by rdb** [comment]
While implementing this, we can also address:
* The fact that the values are int even though OpenGL accepts floats
* Adding the "clamp" parameter supported in OpenGL 4.6 and Vulkan
